### ISSUE TITLE
Auto correct should not automatically remove trailing whitespace in heredoc

### ISSUE THREAD

**Issue by KNejad** [rubocop-hq/rubocop#7860]
Back in 2017 https://github.com/rubocop-hq/rubocop/issues/4517 raised the issue that trailing whitespaces should not be marked as offences in heredoc strings.

This resulted in https://github.com/rubocop-hq/rubocop/pull/5777 which added a new config option to prevent the cop from complaining.

Whilst this seems like a good fix, it still means that when running auto correct, it will remove trailing spaces automatically, which could result in a change in logic.

Take the below for example:

```
require 'digest'

my_string = <<~END
Whitespace here: 
END

sha = Digest::SHA2.new(256).hexdigest my_string
expected_sha = "74788f79010cd7a165e6df2e91013ce72903696bca4d42989de97c3150b89db2"

puts "authenticated" if expected_sha == sha

```

This will print "authenticated" before running `rubocop --auto-correct`, but will fail after.

I've had a similar issue in production where we had to verify a signed string and to test I used an example provided string, which had trailing whitespace. Running `rubocop --auto-correct` broke the tests.

--------

## Expected behavior

Running `rubocop --auto-correct` should not remove the trailing whitespace.

## Actual behavior

The whitespace is removed

## Steps to reproduce the problem

See the script above

## RuboCop version

0.81.0

**Comment by jonas054** [comment]
I agree. I think we chose the wrong default in #5777. It should be
```yaml
Layout/TrailingWhitespace:
  AllowInHeredoc: true
```

**Comment by koic** [comment]
`AllowInHeredoc: true` looks good default value to me. I opened #7883.
